src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -403,9 +403,16 @@ def upload_stream(
             else:
                 raise ex
 
-    def notify_query_listeners(self, query_record: QueryRecord) -> None:
+    def notify_query_listeners(
+        self, query_record: QueryRecord, is_error: bool = False
+    ) -> None:
         with self._lock:
             for listener in self._query_listener:
+                if is_error and not (
+                    listener.session.stored_procedure_profiler._query_history
+                    == listener
```

Comment:
can you help explain what this is checking? it's a little confusing that `_query_history` circles back to `listener`

`listener.session.stored_procedure_profiler._query_history == listener`